docs/03-move/100-move-prover/02-mvp-tutorial.md

```diff
@@ -1,5 +1,606 @@
+Then use `let` and `let post` to define several variables, corresponding to the balances in the two addresses `addr_from` and `to` before and after the function is executed.
+Finally, use the `ensures` statement to tell Prover that the balance in `from` should be subtracted by `amount`; the balance in `to` should be increased by `amount`.
+
+At first glance, there seems to be no problem at all. But is it really so?
+Let's see if Prover thinks this is "the correct description of the behavior of this function".
+After typing `mpm package prove` you can see:
+
+```
+error: post-condition does not hold
+   ┌─ ./sources/BasicCoin.move:58:9
+   │
+58 │         ensures balance_from_post == balance_from - amount;
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │
+   =     at ./sources/BasicCoin.move:45: transfer
+   =     at ./sources/BasicCoin.move:51: transfer (spec)
+   =     at ./sources/BasicCoin.move:53: transfer (spec)
+   =     at ./sources/BasicCoin.move:54: transfer (spec)
+   =     at ./sources/BasicCoin.move:45: transfer
+   =         from = signer{0x0}
+   =         to = 0x0
+   =         amount = 1
+   =         _witness = <generic>
+```
+
+It is unexpectedly unexpected. Prover prompts that the rear conditions are not satisfied, indicating that the behavior described in the previous spec block is not exactly the same as the `transfer` function.
+Why is this so? Let's look down again: the parameters that are not satisfied with the rear conditions are `from = signer{0x0}` and `to = 0x0`. We should know the reason: when the account transfers money to itself, both `to` and `from` point to the same address, so the balance does not change.
+
+There are two solutions now:
+
+**Plan A** does not modify the function definition, but changes the specification.
+In the spec block, consider whether the two accounts for the transfer and receiving are the same address.
+
+```rust
+let post eq_post = balance_to == balance_to_post;
+let post ne_post = balance_from_post == balance_from - amount
+                && balance_to_post   == balance_to   + amount;
+ensures (addr_from == to && eq_post) || (addr_from != to && ne_post);
+```
+
+Or use another slightly more intuitive if syntax:
+
+```rust
+let post eq_post = balance_to == balance_to_post;
+let post ne_post = balance_from_post == balance_from - amount
+                && balance_to_post   == balance_to   + amount;
+ensures if (addr_from == to) eq_post else ne_post;
+```
+
+Note that `if (P) E1 else E2` is not the same as conditional execution in program logic -- it's actually a syntactic sugar equivalent to `ensures` both `P ==> E1` and `!P ==> E2`.
+And `p ==> q` is actually `!p || q`.
+
+That is to say, the end of the second way of writing actually represents this logic:
+
+```rust
+ensures (addr_from == to  ===>  eq_post) && (addr_from != to  ===> ne_post);
+```
+
+scilicet:
```

Comment:
Maybe I'm wrong but I think this is rarely used in technical documents. I suggest to use "that is", "namely", or "equivalently,".